plugins/generator-1.14.4/forge-1.14.4/templates/block.java.ftl

```diff
@@ -71,12 +73,18 @@ public class ${name}Block extends ${JavaModName}Elements.ModElement {
 				${data.blockBase}Block
 			<#else>
 				Block
-			</#if> {
+			</#if>
+			<#if !data.hasGravity && !data.blockBase?has_content && data.isWaterloggable>
+            implements IWaterLoggable
+            </#if> {
 
 		<#if data.rotationMode == 1 || data.rotationMode == 3>
 		public static final DirectionProperty FACING = HorizontalBlock.HORIZONTAL_FACING;
 		<#elseif data.rotationMode == 2 || data.rotationMode == 4 || data.rotationMode == 5>
 		public static final DirectionProperty FACING = DirectionalBlock.FACING;
+        </#if>
+        <#if !data.hasGravity && !data.blockBase?has_content && data.isWaterloggable>

```

Comment:
Instead of doing this (checking for 3 things) every time I could add a listener for waterlogging if its okay?

Reply:
I suggest keeping both to just be sure. 

I would also disable the waterlogging option when the block base is selected.

Reply:
Oh sorry didn't see this comment I'll change them accordingly when I have time